src/oop/user_defined_classes/hard/q3/MainTest.java

```diff
@@ -0,0 +1,78 @@
+package oop.user_defined_classes.hard.q3;
+
+import global.ObjectTest;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static global.tools.CustomAssertions._assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class MainTest {
+    // Java
+    private final String testClassName = "Cupboard";
+    private final String intAttributeName1 = "pots";
+    private final String booleanAttributeName1 = "anyClean";
+    private final String methodName = "washComplete";
+    private ObjectTest classInstance;
+
+    @BeforeEach
+    public void setUp() {
+        String testClassString = "oop.user_defined_classes.hard.q3." + testClassName;
+        classInstance = new ObjectTest(testClassString);
+    }
+
+    @Test
+    public void cupboardClassHasCorrectAttributes() {
+        String missingAttributeMessage =
```

Comment:
We should mention the attribute that is missing or has wrong modifier in the error message.